subsys/bluetooth/host/adv.c

```diff
@@ -535,14 +540,16 @@ static int hci_set_ad_ext(struct bt_le_ext_adv *adv, uint16_t hci_op,
 	}
 
 	if ((total_len_bytes > BT_HCI_LE_EXT_ADV_FRAG_MAX_LEN) &&
-	    atomic_test_bit(adv->flags, BT_ADV_ENABLED)) {
+	    atomic_test_bit(adv->flags, BT_ADV_ENABLED) &&
+	    atomic_test_bit(adv->flags, BT_ADV_EXT_ADV)) {
 		/* It is not allowed to set advertising data in multiple
 		 * operations while the advertiser is running.
 		 */
 		return -EAGAIN;
 	}
 
-	if (total_len_bytes <= BT_HCI_LE_EXT_ADV_FRAG_MAX_LEN) {
+	if (total_len_bytes <= BT_HCI_LE_EXT_ADV_FRAG_MAX_LEN ||
+	    !atomic_test_bit(adv->flags, BT_ADV_EXT_ADV)) {
```

Comment:
I would suggest to remove the additional checks for advertising type when determining if complete or fragmented data will be used. That makes the behavior consistent for legacy and extended advertising when using extended advertising commands.

This approach will disallow setting data with a local name longer than `BT_HCI_LE_EXT_ADV_FRAG_MAX_LEN` while the advertiser is running even if we could have truncated the name. We can perform a dry run to determine if a truncation of advertising name is possible, but IMO this increases the complexity too much.

Another suggestion is to disallow truncating of advertising name when using extended commands and extended advertising type. That would require us to put an additional check inside `hci_set_adv_ext_complete()`